RFC/src/RFC-0241_AtomicSwapXMR.md

```diff
@@ -0,0 +1,823 @@
+![swap flow](assets/TXR_XMR_flow.png)
+
+## Heights and other considerations
+
+It should be noted that when looking at the two lock heights, the first lock height should be sufficiently large enough
+to give ample time for Alice to post the Tari UTXO transaction and for it to be mined with a safe number of confirmations,
+and for Bob to post the Monero transaction and for it to be mined with a safe number of confirmations. The second lock
+height should give ample time for Alice after the first lock height to re-claim her Tari. Larger heights here might make
+refunds slower, but it should be safer in giving more time to finalize this. 
+
+
+Allowing both to claim the Tari after the second lock height is, on face value, a safer option. This can be done by enabling
+either party can claim the script with the lapse transaction. The counterparty can then claim the Monero, however, this
+will open up an attack vector to enable either party to claim the Monero while claiming the Tari. Either party could trivially
+pull off such a scheme by performing a front-running attack and having a bit of luck. The counterparty monitors all broadcast 
+transactions to base nodes. Upon identifying the lapse transaction, they do two things; in quick succession, broadcast 
+their lapse transaction and the transaction to claim the Monero, both with sufficiently high fees. Base nodes will 
+prefer to mine transactions with the higher fees, and thus the counterparty can walk away with both the Tari and the 
+Monero.
+
+
+## Key construction
+
+Using [multi-signatures](https://reyify.com/blog/flipping-the-scriptless-script-on-schnorr) with Schnorr signatures, we
+need to ensure that the keys are constructed so that key cancellation attacks are not possible. To do this, we create new
+keys from the chosen public keys \\(K_a'\\) and \\(K_b'\\)
+
+$$
+\begin{aligned}
+K_a &=  \hash{\hash{K_a' \cat K_b'} \cat K_a' } * K_a' \\\\
+k_a &=  \hash{\hash{K_a' \cat K_b'} \cat K_a' } * k_a' \\\\
+K_b &=  \hash{\hash{K_a' \cat K_b'} \cat K_b' } * K_b' \\\\
+k_b &=  \hash{\hash{K_a' \cat K_b'} \cat K_b' } * k_b' \\\\
+\end{aligned}
+\tag{1}
+$$
+
+## Method 1
+
+### Detail
+
+This method relies purely on TariScript to enforce the exposure of the private Monero aggregate keys. The script forces 
+the spending party to supply their Monero private key part as input data to the script, evaluated via the operation 
+`Ed25519Point`. This TariScript operation will reveal part of the aggregated Monero private key publicly but this is still secure:
+see [Key security](#key-security).
+
+The simplicity of this method lies therein that the spending party creates all transactions on their 
+own. Bob requires a pre-image from Alice to complete the swap transaction; Alice needs to verify that Bob published the 
+Monero transaction and that everything is complete as they have agreed. If she is happy, she will provide Bob with the 
+pre-image to immediately claim the Tari UTXO.
+
+### Key security
```

Comment:
This section can probably be removed entirely

Reply:
I think it is very informative to show this for a mixed audience, even though it might seem trivial. Many people wonder why it is safe to construct an aggregated public key where their counterparty knows the other portion of the aggregated secret key.

Reply:
I have made my viewpoints public above that I don't feel this contributes much to the article. We need to make a call about do we keep this or not.

Reply:
Fair enough, we can keep it. Ideally it should just be somewhere else, with a link